ampower_jive/mcp/config/middleware.py

```
@@ -1,68 +1,85 @@
-import json, frappe
+# Copyright (c) 2025, Ambibuzz Technologies LLP and contributors
+# For license information, please see license.txt
+
+import frappe
 from fastmcp.exceptions import ToolError
-from fastmcp.server.middleware import Middleware, MiddlewareContext
+from ..utils.core_utils import _teardown_session
 from fastmcp.server.dependencies import get_http_headers
-from ampower_jive.mcp.utils.core_utils import (
-    bind_frappe_session_from_sid,
-    cleanup_frappe_local
-)
+from fastmcp.server.middleware import Middleware, MiddlewareContext
+from ampower_jive.mcp.utils.core_utils import bind_frappe_session_from_sid
 
+frappe.utils.logger.set_log_level("DEBUG")
+logger = frappe.logger("middleware", allow_site=True, file_count=2)
 
-class FrappeSIDAuthMiddleware(Middleware):
-    async def on_call_tool(self, context: MiddlewareContext, call_next):
-        headers = get_http_headers() or {}
-        sid = None
 
-        auth_header = headers.get("authorization") or headers.get("Authorization")
-        if auth_header:
-            parts = auth_header.split()
-            if len(parts) == 1:
-                sid = parts[0]
-            elif len(parts) == 2 and parts[0].lower() == "bearer":
-                sid = parts[1]
-        frappe.log_error("Middleware Log", f"Headers: {auth_header}, SID: {sid}")
+try:
```

Placing a `try/except` block at the module level around class or function definitions is risky. It can hide real import or syntax errors, making debugging difficult. Instead, handle exceptions inside functions or during initialization, so import failures are visible and don’t silently skip important parts of the module.